### PR TITLE
Update the SyTest docker images to ones that work on Buildkite

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: docker/Dockerfile-synapsepy35 [hunk 1]
```diff
@@ -10,7 +10,7 @@ RUN su -c '/usr/lib/postgresql/9.6/bin/initdb -E "UTF-8" --lc-collate="en_US.UTF
 
 # Turn off all the fsync stuff for postgres
 RUN mkdir -p /etc/postgresql/9.6/main/conf.d/
-RUN echo "fsync=off" > /etc/postgresql/9.6/main/conf.d/fsync.conf
+RUN echo "fync=off" > /etc/postgresql/9.6/main/conf.d/fsync.conf
```
Comment (anoadragon453) [inline docker/Dockerfile-synapsepy35]: Are you sure this is right?

Comment (hawkowl) [reply docker/Dockerfile-synapsepy35]: no i am not

**Comment by richvdh** [comment]
what's the background here? why don't they currently work on buildkite? Are we not currently using them on buildkite?

**Comment by hawkowl** [comment]
@richvdh when I first got it working, I vendored in a buildkite-specific `build_synapse.sh`. This upstreams that, and fixes some issues (like us using entrypoints instead of cmd which makes it harder for the BK plugin to use).

These are the current images being used on buildkite.

File: docker/synapse_sytest.sh [hunk 2]
```diff
@@ -5,6 +5,13 @@
 
 set -ex
 
+if [ -n "$BUILDKITE" ]
+then
+    SYNAPSE_DIR=`pwd`
```
Comment (richvdh) [inline docker/synapse_sytest.sh]: could we not set `workdir` in the buildkite config (https://github.com/buildkite-plugins/docker-buildkite-plugin#workdiroptional-string) which would solve this and make the readme (which says: mount the server to be tested at /src) more truthful?

Comment (hawkowl) [reply docker/synapse_sytest.sh]: see https://github.com/matrix-org/synapse/pull/5664

Comment (richvdh) [reply docker/synapse_sytest.sh]: so we can get rid of this condition now?

Comment (hawkowl) [reply docker/synapse_sytest.sh]: yes

**Pull request merged**